qemu/tests/virtio_driver_sign_check.py

```diff
@@ -1,85 +1,71 @@
-    au3_link = os.path.join(test.bindir, au3_link)
-    signtool_cmd = params.get("signtool_cmd")
-    list_files_cmd = params.get("list_files_cmd")
-    drive_list = params.get("drive_list")
-    vm.copy_files_to(au3_link, "c:\\")
-    drivers = {}
-    logging.info("Install sdk in guest if signtool is not available.")
-    session.cmd_status_output(signtool_install, timeout=360)
-    # Wait until guest start install sdk.
-    time.sleep(10)
-    logging.info("Waiting for guest sdk setup ...")
-    utils_misc.wait_for(is_sdksetup_finished, timeout=1800)
-    results_all = """All the signature check log:\n"""
-    fails_log = """Failed signature check log:\n"""
-    fail_num = 0
-    fail_drivers = []
-    logging.info("Running SignTool command in guest...")
-    try:
-        for drive in drive_list.split():
-            for type in ['.cat', '.sys']:
-                driver = session.cmd_output(list_files_cmd % (drive, type)).\
-                    splitlines()[1:-1]
-                drivers[type] = driver
-            files = zip(drivers['.cat'], drivers['.sys'])
-            for cat, sys in files:
-                cmd = signtool_cmd % (cat, sys)
-                s, result = session.cmd_status_output(cmd)
-                if s:
-                    msg = "Fail command: %s. Output: %s" % (cmd, result)
-                    test.fail(msg)
-                results_all += result
-                re_suc = "Number of files successfully Verified: ([0-9]*)"
-                try:
-                    suc_num = re.findall(re_suc, result)[0]
-                except IndexError:
-                    msg = "Fail to get Number of files successfully Verified"
-                    test.fail(msg)
-
-                if int(suc_num) != 1:
-                    fails_log += result
-                    fail_num += 1
-                    fail_driver = cat + " " + sys
-                    fail_drivers.append(fail_driver)
-        if fail_num > 0:
-            msg = "Following %s driver(s) signature checked failed." % fail_num
-            msg += " Please refer to fails.log for details error log:\n"
-            msg += "\n".join(fail_drivers)
-            test.fail(msg)
+    signtool_cmd = params["signtool_cmd"]
+    list_files_cmd = params["list_files_cmd"]
 
+    try:
+        error_context.context("Running SignTool check test in guest...",
+                              logging.info)
+        file_type = [".cat", ".sys", ".inf", "Wdf"]
+        tested_list = []
+        viowin_letter, path = get_driver_file_path(session, params)
+        for ftype in file_type:
+            cmd = list_files_cmd % (viowin_letter, path, ftype)
+            driver_file = session.cmd_output(cmd, timeout).splitlines()[:-1]
+            tested_list.extend(driver_file)
+        for driver_file in tested_list[1:]:
+            test_cmd = signtool_cmd % (tested_list[0], driver_file)
+            test_cmd = utils_misc.set_winutils_letter(session, test_cmd)
```

Comment:
Better to move this line out from the loop, otherwise it will look up winutils letter several times.

Reply:
yes, you are right, have updated~
thanks~